engine/src/main/java/com/datatorrent/stram/util/AbstractWritableAdapter.java

```diff
@@ -56,6 +63,9 @@ public void readFields(DataInput arg0) throws IOException
       ois.close();
     }
     catch (Exception e) {
+      final Path path = Files.createTempFile("dt-", ".ser");
+      logger.error("Failed to de-serialize {}. Writing raw data to {}.", this.getClass().getName(), path, e);
```

Comment:
The user will not know that this data will be logged, unlike checkpoints (where he can control how data gets saved), but it's ok to address this in a broader context. It would be good to keep track of this in some document, all the points where we touch disk, buffer spooling, tuple recording etc.